core/src/main/java/org/elasticsearch/index/store/StoreFileMetaData.java

```diff
@@ -38,6 +36,8 @@
 
     public static final Version FIRST_LUCENE_CHECKSUM_VERSION = Version.LUCENE_5_0_0;
 
+    public static final String UNKNOWN_CHECKSUM = "_na_";
```

Comment:
I'll add a method to `StoreFileMetaData` with something to the effect of `isUnknownChecksum`, then this field can be `private`.  Its still helpful to have a central place for it b/c other places use it (i.e. check on "_na_")